MSAL/test/integration/native_auth/end_to_end/reset_password/MSALNativeAuthResetPasswordEndToEndTests.swift

```diff
@@ -155,4 +126,26 @@ final class MSALNativeAuthResetPasswordEndToEndTests: MSALNativeAuthEndToEndBase
         XCTAssertNotNil(signInAfterResetPasswordDelegate.result?.idToken)
         XCTAssertNotNil(signInAfterResetPasswordDelegate.result?.account.accountClaims)
     }
+
+    func submitCode(resetPasswordStartDelegate: ResetPasswordStartDelegateSpy, username: String, retries: Int) async -> ResetPasswordRequiredState? {
+        let passwordRequiredExp = expectation(description: "password required")
```

Comment:
should not check the retries that it is greater that 0 as first action?

Reply:
I check in the line below, before calling:
`if resetPasswordVerifyDelegate.onResetPasswordVerifyCodeErrorCalled && resetPasswordVerifyDelegate.error?.isInvalidCode == true && retries > 0 `
Technically I could move it there but that would be one extra function call which returns nil

Reply:
We are re-sending  the code up to 3 times if we get the error "invalid code" from the server with this check:
`resetPasswordVerifyDelegate.onResetPasswordVerifyCodeErrorCalled && resetPasswordVerifyDelegate.error?.isInvalidCode == true`. This caters for the case that happens most often, the email is never sent.
At the same time we are retrying 3 times to get the email from the 1SecMail API. This caters for the 2-nd most common issue - email arrives after a rather big delay (more than the 4seconds at the start)